contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/data/rest/webmvc/HalFormsProfileControllerTest.java

```diff
@@ -0,0 +1,170 @@
+                                        {
+                                            name: "content.mimetype",
+                                            type: "text"
+                                        },
+                                        {
+                                            name: "content.filename",
+                                            type: "text"
+                                        },
+                                        {
+                                            name: "orders",
+                                            type: "url",
+                                            options: {
+                                                valueField: "/_links/self/href",
+                                                minItems: 0,
+                                                link: {
+                                                    href: "http://localhost/orders"
+                                                }
+                                            }
+                                        },
+                                        {
+                                            name: "invoices",
+                                            type: "url",
+                                            options: {
+                                                valueField: "/_links/self/href",
+                                                minItems: 0,
+                                                link: {
+                                                    href: "http://localhost/invoices"
+                                                }
+                                            }
+                                        }
+                                    ]
+                                }
+                            }
+                        }
+                        """));
+    }
+
+    @Test
+    void profileController_requiredAssociation() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/profile/invoices")
+                        .accept(MediaTypes.HAL_FORMS_JSON)
+                ).andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.content().json("""
+                    {
+                        _templates: {
+                            "create-form": {
+                                method: "POST",
+                                contentType: "application/json",
+                                target: "http://localhost/invoices",
+                                properties: [
+                                    {
+                                        name: "number",
+                                        required: true,
+                                        type: "text"
+                                    },
+                                    {
+                                        name: "draft"
+                                        # ,type: "checkbox"
```

Comment:
commented out ?

Reply:
Yes, I want those types there, but unfortunately spring-hateoas does not automatically set the type to checkbox for boolean fields.